experimental/rocm/native_executable.c

```diff
@@ -74,22 +74,30 @@ iree_status_t iree_hal_rocm_native_executable_create(
       entry_count * sizeof(iree_hal_pipeline_layout_t*);
   iree_status_t status = iree_allocator_malloc(context->host_allocator,
                                                total_size, (void**)&executable);
-  executable->pipeline_layouts =
-      (void*)((char*)executable + sizeof(*executable) +
-              entry_count * sizeof(iree_hal_rocm_native_executable_function_t));
-  hipModule_t module = NULL;
   if (iree_status_is_ok(status)) {
+    executable->context = context;
+    iree_hal_resource_initialize(&iree_hal_rocm_native_executable_vtable,
+                                 &executable->resource);
```

Comment:
iree_hal_resource_initialize usually goes first
```suggestion
    iree_hal_resource_initialize(&iree_hal_rocm_native_executable_vtable,
                                 &executable->resource);
    executable->context = context;
```

Reply:
I changed the order.